File: src/Feline.java
public class Feline extends Animal{

    Feline(){
        super(false, "Carnivore", "unknown", true, "Felix", 4);
    }

    /**
     *  Constructor with only Animal parameters
     * @param awake:boolean - Is the animal awake?
     * @param diet:String - What does the animal eat?
     * @param habitat:String - Where does the animal live?
     * @param hungry:boolean - Is the animal hungry?
     * @param name:String - The name of the animal
     * @param numLegs:int - How many legs does the animal have?
     */
    Feline(boolean awake, String diet, String habitat, boolean hungry, String name, int numLegs){
        super(awake, diet, habitat, hungry, name, numLegs);
    }

    /**
     *  Override for roam function
     *  there is a 50% chance for the Feline to either stroll around the enclosure or go to sleep
     */
    @Override
    public void roam() {
        double randomNumber = Math.random();
        if(randomNumber < 0.5) {
            System.out.println(name + " the " + this.getClass().getName() + ": Began to stroll around the enclosure.");
        }else{
            System.out.println(name + " the "  + this.getClass().getName() + ": Looked at you and went to sleep.");
        }
    }

    /**
     *  Override for makeNoise function
     *  Just has the Feline meow
     */
    @Override
    public void makeNoise() {
        System.out.println(name + " the " + this.getClass().getName() + ": Meowed");
    }
}

File: Project2/Feline.py
import random
from Animal import Animal
from DevourBehavior import DevourBehavior

class Feline(Animal):

  '''
    Generic initializer for Feline
    @param name:String, default:'Flower'
    @param awake:Boolean, default:False
    @param hungry:Boolean, deault:True
    @param num_legs:Int, default:4
    @param eat_behavior:EatBehavior, default:DevourBehavior
  '''
  def __init__(self, name='Flower', awake=False, hungry=True, num_legs=4, eat_behavior=DevourBehavior):
    super(Feline, self).__init__(name, awake, hungry, num_legs, eat_behavior)

  '''
    Feline has a 25% chance to jump around and 75% chance to knock over a mug
  '''
  def roam(self):
    animal_type = type(self).__name__
    random_number = random.random()
    if random_number < 0.25 :
      print('{} the {} jumped around.'.format(self.name, animal_type))
    else:
      print('{} the {} knocked over a mug.'.format(self.name, animal_type))

  '''
    Feline meows
  '''
  def make_noise(self):
    animal_type = type(self).__name__
    print('{} the {} meowed'.format(self.name, animal_type))

